docs/application/dotnet/guides/nui/visuals.md

```diff
@@ -1,390 +1,552 @@
-private const int ColorVisualPropertyIndex = PROPERTY_REGISTRATION_START_INDEX + 1;
-private const int PrimitiveVisualPropertyIndex = PROPERTY_REGISTRATION_START_INDEX + 2;
+   ```
+   ColorVisual _colorVisual_1 = new ColorVisual();
 
-private VisualBase _colorVisual;
+   // obligatory properties
+   _colorVisual_1.Color = Color.Red;
 
-PropertyMap colorVisual = new PropertyMap();
-colorVisual.Add(Visual.Property.Type, new PropertyValue((int)Visual.Type.Color))
-           .Add(ColorVisualProperty.MixColor, new PropertyValue(_color));
-_colorVisual = VisualFactory.Get().CreateVisual(colorVisual);
+   // optional properties:
+   _colorVisual_1.Size = new Size2D(400,400);
+   _colorVisual_1.DepthIndex = 5;
 
-RegisterVisual(ColorVisualPropertyIndex, _colorVisual);
+   _visualView.AddVisual("RedRectangle", _colorVisual_1);
 
-/// Set the color visual depth index
-_colorVisual.DepthIndex = ColorVisualPropertyIndex;
-```
 
-<a name="gradientvisual"></a>
-## Gradient Visual
+   ColorVisual _colorVisual_2 = new ColorVisual();
+
+   // obligatory properties
+   _colorVisual_2.Color = new Color(0.0f, 0.0f, 1.0f, 0.8f);
+
+   // optional properties:
+   _colorVisual_2.Size = new Size2D(300,300);
+   _colorVisual_2.Origin = Visual.AlignType.BottomEnd;
+   _colorVisual_2.AnchorPoint = Visual.AlignType.BottomEnd;
+   // shifting the visual left and up 
+   _colorVisual_2.RelativePosition = new RelativeVector2(-0.1f, -0.1f);
+   // for values > 0 corners are rounded
+   _colorVisual_2.CornerRadius = 50.0f;
+   // setting a value lower than for the previous object causes it to be covered
+   // without changing this value, the objects are drawn in the order they were added
+   _colorVisual_2.DepthIndex = _colorVisual_1.DepthIndex - 1;
+
+   _visualView.AddVisual("BlueRectangle", _colorVisual_2);
+   ```
+
+
+**Figure: ColorVisual obtained from the above code (_visualView background color set to gray).**
+
+ | default values of the `DepthIndex`                   | changed `DepthIndex` values                                          |
+ |------------------------------------------------------|----------------------------------------------------------------------|
+ | ![ColorVisual](media/visuals/ColorVisualExample.png) | ![ColorVisualDepth](media/visuals/ColorVisualExample_DephtIndex.png) |
 
-The gradient visual renders a smooth transition of colors to the control's quad. Both linear (left in the following figure) and radial (right in the following figure) gradients are supported.
 
-**Figure: Gradient visual**
 
-![Linear gradient visual](media/linear-gradient-visual.png) ![Radial gradient visual](media/radial-gradient-visual.png)
+<!---------------------------------------------------------------------------------------------------------------------------->
+<a name="gradientVisual"></a>
+### GradientVisual
+
+The GradientVisual renders a smooth transition of colors to the control. NUI supports both, linear and radial gradients.
```

Comment:
The GradientVisual renders a smooth transition of colors to the control.

Not able to understand this what do you want to convey here

Reply:
I wanted to emphasize that the GradientVisual renders not a solid color but a smooth transition between multiple colours.